### PR TITLE
Release branch merge master

### PR DESCRIPTION
**Pull request by rayliangatos**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###
No changes, just sync master

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by Mo-Lala84 — DISMISSED** [review]
approved

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -3,8 +3,30 @@
 ### Version 4.0.0-activity-tracker-unauthorised-beta
 **EUI-4226** Fixed an issue with 401/403 errors ignoring that the user is unauthorised.
 
+### Version 3.1.5-complex-hidden-collection-show
```
Comment (paul-graham) [inline RELEASE-NOTES.md:6]: _Nitpick:_
Do we want to sort out the order of versions in here? We've got 4.1.0 coming after a bunch of earlier ones...

Comment (paul-graham) [reply RELEASE-NOTES.md:6]: And a bunch of older versions deleted...? 🤔

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@hmcts/ccd-case-ui-toolkit",
-  "version": "4.0.0-activity-tracker-unauthorised-beta",
+  "version": "4.1.6",
```
Comment (paul-graham) [inline package.json:3]: There should be an entry for this version in `RELEASE-NOTES.md`.

**Review by RiteshHMCTS — DISMISSED** [review]
approved

File: src/shared/components/palette/datetime-picker/datetime-picker.component.html [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
   <fieldset>
     <legend>
       <span class="form-label" *ngIf="caseField.label">{{caseField | ccdFieldLabel }}</span>
       <span class="form-hint" *ngIf="caseField.hint_text">{{caseField.hint_text}}</span>
       <span class="error-message"
-            *ngIf="dateControl && dateControl.errors && dateControl.touched && !(this.minError || this.maxError)">{{dateControl.errors | ccdFirstError:caseField.label}}</span>
+            *ngIf="dateControl && dateControl.errors && dateControl.dirty && !(this.minError || this.maxError)">{{dateControl.errors | ccdFirstError:caseField.label}}</span>
             <span class="error-message"
-            *ngIf="dateControl && dateControl.touched && this.minError">This date is older than the minimum date allowed</span>
+            *ngIf="dateControl && dateControl.dirty && this.minError">This date is older than the minimum date allowed</span>
```
Comment (paul-graham) [inline src/shared/components/palette/datetime-picker/datetime-picker.component.html:10]: _Super picky:_
```suggestion
            *ngIf="dateControl && dateControl.dirty && this.minError">This date is earlier than the earliest date allowed</span>
```

File: src/shared/components/palette/datetime-picker/datetime-picker.component.html [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
             <span class="error-message"
-            *ngIf="dateControl && dateControl.touched && this.maxError">This date is later than the maximum date allowed</span>
+            *ngIf="dateControl && dateControl.dirty && this.maxError">This date is later than the maximum date allowed</span>
```
Comment (paul-graham) [inline src/shared/components/palette/datetime-picker/datetime-picker.component.html:12]: _Super picky:_
```suggestion
            *ngIf="dateControl && dateControl.dirty && this.maxError">This date is later than the latest date allowed</span>
```

File: src/shared/components/palette/palette.module.ts [hunk 5]
```diff
@@ -16,7 +16,7 @@ import { ComplexModule } from './complex/complex.module';
 import { AddressModule } from './address/address.module';
 import { BaseFieldModule } from './base-field/base-field.module';
 import { WriteTextFieldComponent } from './text/write-text-field.component';
-import { ReactiveFormsModule } from '@angular/forms';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
```
Comment (paul-graham) [inline src/shared/components/palette/palette.module.ts:19]: _Nitpick:_
Doesn't look like `FormsModule` is being used anywhere, which would make this import redundant.

File: RELEASE-NOTES.md [hunk 6]
```diff
@@ -3,8 +3,30 @@
 ### Version 4.0.0-activity-tracker-unauthorised-beta
```
Comment (paul-graham) [inline RELEASE-NOTES.md:3]: Where's the entry for `4.1.6`, which is the version mentioned in `package.json`?

**Review by paul-graham — CHANGES_REQUESTED** [review]
Mostly _nitpicks_, but `RELEASE-NOTES.md` is missing an entry for the new version set up in `package.json` and the file itself is out of sequence. There are also a lot of entries being removed, which is a bit puzzling...

File: RELEASE-NOTES.md (hunk 6 above)
Comment (rayliangatos) [reply RELEASE-NOTES.md:3]: this is to sync the master->Release, I have updated the version to 4.1.6-sync-master now, as this is a PR requrested by Mo to resolve conflicts merge from master->Release, personally I just took what is in master as priority, and then take the latest changes from Release by compare both branches, if it's not accurate enough on merging stuff from Release branch, as I don't have an accurate answer on some specific conflicts from Release branch, and it needs to go back the specific developer to resolve the merge conflicts regarding Release changes.

File: RELEASE-NOTES.md (hunk 1 above)
Comment (Mo-Lala84) [reply RELEASE-NOTES.md:6]: Will be cleaned up as part of RELEASE PREPS

File: src/shared/components/palette/datetime-picker/datetime-picker.component.html (hunk 3 above)
Comment (Mo-Lala84) [reply src/shared/components/palette/datetime-picker/datetime-picker.component.html:10]: We will tackle this in IP week

File: src/shared/components/palette/datetime-picker/datetime-picker.component.html (hunk 4 above)
Comment (Mo-Lala84) [reply src/shared/components/palette/datetime-picker/datetime-picker.component.html:12]: We will tackle this in IP week

File: package.json (hunk 2 above)
Comment (Mo-Lala84) [reply package.json]: will be made yes-once this is commited into master

**Review by Mo-Lala84 — APPROVED** [review]
approved

File: src/shared/components/palette/datetime-picker/datetime-picker.component.html (hunk 4 above)
Comment (paul-graham) [reply src/shared/components/palette/datetime-picker/datetime-picker.component.html:12]: Okay, cool. I'll raise a JIRA for this and these date errors...

Comment (paul-graham) [reply src/shared/components/palette/datetime-picker/datetime-picker.component.html:12]: https://tools.hmcts.net/jira/browse/EUI-4332

File: src/shared/components/palette/datetime-picker/datetime-picker.component.html (hunk 3 above)
Comment (paul-graham) [reply src/shared/components/palette/datetime-picker/datetime-picker.component.html:10]: https://tools.hmcts.net/jira/browse/EUI-4332

**Pull request merged**